src/components/App/Header/HeaderComponents/SearchField.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState, useEffect, useRef } from 'react';
+import { useHistory } from "react-router-dom";
+import styled from 'styled-components';
+
+const SrchDivMain = styled.div`
+  @media (min-width: 600px) {
+    width: auto;
+    margin-left: 24px;
+  }
+  width: 100%;
+  position: relative;
+  margin-left: 0;
+  margin-right: 16px;
+  border-radius: 4px;
+  background-color: rgba(255, 255, 255, 0.15);
+`;
+
+const SrchImgCont = styled.div`
+  height: 100%;
+  display: flex;
+  padding: 0px 16px;
+  position: absolute;
+  align-items: center;
+  pointer-events: none;
+  justify-content: center;
+`;
+
+const SrchImg = styled.svg`
+  fill: currentColor;
+  width: 1em;
+  height: 1em;
+  display: inline-block;
+  font-size: 1.5rem;
+  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
+  flex-shrink: 0;
+  user-select: none;
+  svg:not(:root) {
+    overflow: hidden;
+  }
+`;
+
+const SrchCont = styled.div`
+  color: inherit;
+  cursor: text;
+  display: inline-flex;
+  position: relative;
+  font-size: 1rem;
+  box-sizing: border-box;
+  align-items: center;
+  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
+  font-weight: 400;
+  line-height: 1.1876em;
+  letter-spacing: 0.00938em;
+`;
+
+const SrchInput = styled.input`
+  width: 20ch;
+  padding: 8px 8px 8px 0px;
+  transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
+  padding-left: calc(1em + 32px);
+  font: inherit;
+  color: currentColor;
+  border: 0;
+  height: 1.1876em;
+  margin: 0;
+  display: block;
+  min-width: 0;
+  background: none;
+  box-sizing: content-box;
+  animation-name: mui-auto-fill-cancel;
+  letter-spacing: inherit;
+  animation-duration: 10ms;
+  -webkit-tap-highlight-color: transparent;
+  ::placeholder {
+    color: rgba(255, 255, 255, 0.3);
+  }
+`;
+
+function MySearch(props) {
+  const inputText = useRef();
+  const [value, setValue] = useState('');
+  const history = useHistory();
+
+  useEffect(() => {

```

Comment:
I do not get this `useEffect` at all. I think that if you move the onChange to the App component (as I suggested) then this component will be rendered with the change, so won't need any useEffect here.